### DISCUSSION TITLE
Use title as caption

### DISCUSSION THREAD

**Discussion by LITUATUI** [razonyang/hugo-theme-bootstrap#573] [Ideas]
Considering that in markdown we can't define a caption for an image, some themes already use the title field to put captions on images.

`
![Alt](/path/to/img.jpg “image title”)
`

[Example](https://neonmirrors.net/post/2020-10/deploying-harbor-on-photon-os/)

Would be great to have this feature on this theme!

**Comment by Natenom** [comment]
"Image title" is the title of an image and not the caption :) Currently this theme uses your mentioned syntax to set the image title and I think it should stay like that because thats how default markdown in Hugo works.

For anything more than "title", "filename" and "alt" you can use the builtin shortcode **figure**, which does have caption, link, etc...:

```{{< figure caption="supports markdown ..." link="..." src="..." >}}```

https://gohugo.io/content-management/shortcodes/#figure

**Comment by LITUATUI** [comment]
Thanks for the reply. 

I'm converting a big WordPress blog to Hugo and most themes I tried do this. The image title field in markdown is used to set the figcaption element in HTML.

[This](https://m3upt.com/en/) is a website I made with the [DoIt theme](https://github.com/HEIGE-PCloud/DoIt).

Captions are displayed, just by defining a title for the image in markdown.

`![Kodi screenshot](/images/Kodi.avif "Kodi screenshot")
`
`![ProgTV screenshot](/images/ProgTV-RTP1.avif "ProgTV screenshot")
`
`![Perfect Player screenshot](/images/Perfect-Player-Euronews.avif "Perfect Player screenshot")
`

**Comment by Natenom** [comment]
I converted my blog with ~2600 posts half a year ago and had some trouble with this topic, too. I asked myself why developers of Hugo would not implement such simple things like image captions into "their" markdown syntax. I started doing that myself and ended up with something that was not intuitive and that would not work anymore if I switched my blog theme. Then I realized how it works: Do the very basic stuff with a simple Markdown syntax and use shortcodes if you need more.

So I use Markdown syntax for just images (+ alt text) and link them, if needed, with Markdown, too. If I need captions, width, height, ... I use the shortcode figure.

And I think that a theme should be consistent with the Markdown syntax and not use data for something it was not created for, i. e. using "title" as "image caption".

@razonyang If you consider implementing this please add a setting to use the default behavior "title is title" XOR "title becomes caption". :)

**Comment by razonyang** [comment]
@LITUATUI  @Natenom Just created a parameter `post.imageTitleAsCaption` which is disabled by default.

Docs: https://hbs.razonyang.com/v1/en/docs/image-processing/#caption

**Comment by LITUATUI** [comment]
Thanks @razonyang

For some reason, it's not working here. I'll try to find out why.

**Reply by razonyang** [reply]
The feature was merged into `master` branch, it's not released yet. I'll draft a new release ASAP.

**Reply by razonyang** [reply]
FYI, just drafted a new release [v1.0.0-alpha.0](https://github.com/razonyang/hugo-theme-bootstrap/releases/tag/v1.0.0-alpha.0)

**Reply by LITUATUI** [reply]
It works great, thanks!
